app/controllers/bankdetails/SummaryController.scala

```diff
@@ -16,63 +16,52 @@
 
 package controllers.bankdetails
 
-import javax.inject.{Inject, Singleton}
-
 import config.AMLSAuthConnector
 import connectors.DataCacheConnector
 import controllers.BaseController
-import forms.EmptyForm
+import javax.inject.{Inject, Singleton}
 import models.bankdetails.BankDetails
-import models.status.{NotCompleted, SubmissionReady}
-import services.StatusService
 import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
-import utils.StatusConstants
+import utils.RepeatingSection
 
 import scala.concurrent.Future
 
 @Singleton
 class SummaryController @Inject()(
                                    val dataCacheConnector: DataCacheConnector,
-                                   val authConnector: AuthConnector = AMLSAuthConnector,
-                                   val statusService: StatusService
-                                 ) extends BaseController {
+                                   val authConnector: AuthConnector = AMLSAuthConnector
+                                 ) extends BaseController with RepeatingSection {
 
-  private def updateBankDetails(bankDetails: Option[Seq[BankDetails]]) : Future[Option[Seq[BankDetails]]] = {
+  private def updateBankDetails(bankDetails: Option[Seq[BankDetails]], index: Int) : Future[Option[Seq[BankDetails]]] = {
     bankDetails match {
       case Some(bdSeq) => {
-        val updatedList = bdSeq.map { bank =>
-          bank.copy(hasAccepted = true)
+        val updatedList = bdSeq.zipWithIndex.map {
+          case (bank, i) => if (i == index -1) bank.copy(hasAccepted = true) else bank
```

Comment:
We can use `RepeatingSection.updateData` here to update this instead of working out the index ourselves.